config/v1/types_ingress.go

```diff
@@ -152,8 +153,55 @@ type AWSIngressSpec struct {
 	// +kubebuilder:validation:Enum:=NLB;Classic
 	// +kubebuilder:validation:Required
 	Type AWSLBType `json:"type,omitempty"`
+
+	// networkLoadBalancerParameters holds configuration parameters for an AWS
+	// network load balancer. Present only if type is NLB.
+	//
+	// +openshift:enable:FeatureGate=InstallEIPForDefaultNLBIngressController
+	// +optional
+	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancerParameters,omitempty"`
 }
 
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer. For Example: Setting AWS EIPs https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+type AWSNetworkLoadBalancerParameters struct {
+	// eipAllocations is a list of IDs for Elastic IP (EIP) addresses that
+	// are assigned to the Network Load Balancer.
```

Comment:
The Ingress Config API in `spec.loadBalancer` has fields that default for both the `default` IngressController, and/or user-created IngressControllers. @JoelSpeed has previously suggested to split the API for each of these purposes/functions, I agree that would be much more explicit and clear.

However, we can't reasonably expect you to fix all of this now, and if you add a new API field without deprecating and relocating the others, I think the API will become even more confusing. Correct me if you disagree @JoelSpeed. I think we plow ahead with the current design, knowing it's not perfect, but it's hopefully something we can fix later with a V2 or a deprecation/restructuring effort.

All of this to say: we need to document what the purpose of this field is. If we can't have a explicit API structure, it should at least be well documented in the API:

```suggestion
	// eipAllocations is a list of IDs for Elastic IP (EIP) addresses that
	// are assigned to the default IngressController's NLB.
```

Reply:
Well, actually, I might be overstating the effort here. At this current moment, there is actually only two field (`spec.loadBalancer.platform.aws.type` and `spec.domain`) that set "defaults" or the `default` IngressController values.

But still, I don't quite know exactly what to suggest to structure this API. Maybe I'll get @JoelSpeed's feedback tomorrow in API review office hours about this.